articles/word-cloud/index.md

```diff
@@ -0,0 +1,117 @@
+
+Once installed, check if the packages are installed correctly. Run the following piece of code in your terminal, and you should get a valid output for the version number.
+```py
+>>> import nltk
+>>> print(ntlk.__version__)
+>>> import matplotlib
+>>> print(matplotlib.__version__)
+>>> import wordcloud
+>>> print(wordcloud.__version__)
+```
+If you get valid output, you have successfully installed the package and can proceed with the rest of the article. I get the following output: 
+
+```py
+>>> import matplotlib
+>>> matplotlib.__version__
+'3.3.3'
+>>> import nltk
+>>> nltk.__version__
+'3.5'
+>>> import wordcloud
+>>> wordcloud.__version__
+'1.8.1'
+```
+
+### Word Cloud Generation
+
+Let us now look at the code to generate word clouds. The input to the program will be a paragraph copied from any website of your choice. With the paragraph as input, we will pre-process it and send it to the `wordcloud` package. Let's begin.
+
+As mentioned above, we use the following libraries:
+1. **[matplotlib](https://www.section.io/engineering-education/matplotlib-visualization-python/):** A visualization and plotting tool used extensively in Python. 
+2. **[nltk.corpus.stopwords](https://pythonprogramming.net/stop-words-nltk-tutorial/):** Natural language toolkit, known as `nltk,` is a library built for performing various Natural Language Processing (NLP) tasks. It is a vast library with tools for pre-processing, data cleaning, data visualization, data modeling, etc. We will use the list of stopwords for English. Stopwords are redundant words that don't add significant meaning to the data.
+3. **[nltk.tokenize.word_tokenize](https://www.nltk.org/api/nltk.tokenize.html):** Tokenization is the process of breaking down the text into smaller units called tokens. The tokens can be words, sub-words, or phrases. We will use the tokenizer available in `nltk.` 
+4. **[wordcloud](https://pypi.org/project/wordcloud/):** It is a library that takes in the list of words and outputs a word cloud image. Developed by [Andrea Mueller](https://amueller.github.io/), it is quite extensible and flexible with respect to the features.
+
+#### Code
+We define a class called `WordCloudGeneration` and define the following methods in the class:
+1. ** pre-processing**: We pass the input `data` through the tokenizer. The `data` is converted to lower case and tokenized. Tokenization results in a list of words. This list of words is further filtered. The filtering process copies words to `preprocessed_data` only if the word is not a stopword.
+
+2. **create_word_cloud**: This function takes in the processed list of words and calls the `WordCloud` class object. The `generate` method in the `WordCloud` class returns an image of the word cloud. Using the library `matplotlib`, we plot the image. 
+
+
+```py
+import matplotlib.pyplot as plt
+
+from nltk.corpus import stopwords
```

Comment:
To be able to use this line the reader should have downloaded the `stopwords` and `punkt` packages using the [nltk downloader](https://www.nltk.org/data.html)
You should add that step to the installation.

Reply:
I created a new setup for this tutorial. There was no such requirement on my end. Can you tell me what the error is.